src/main/webapp/app/entities/submission.model.ts

```diff
@@ -62,6 +62,18 @@ export function getLatestSubmissionResult(submission: Submission | undefined): R
     }
 }
 
+/**
+ * Used to access a submissions result for a specific correctionRound
+ *
+ * @param submission
+ * @returns the results or undefined if submission or the result for the requested correctionRound is undefined
+ */
+export function getSubmissionResultByCorrectionRound(submission: Submission | undefined, correctionRound: number): Result | undefined {
+    if (submission?.results) {
```

Comment:
maybe add ` && submission?.results.length >= correctionRound` just to make sure we're not out of bounds?

Reply:
I think they are using the fact that js has no out of bounds errors here. JavaScript arrays don’t really have a concept of size. You can try to retrieve an element from any position in the array, and if no element exists it will return undefined.

Reply:
yes, will add that!